### PR TITLE
Add extras in kafka producer

### PR DESCRIPTION
**Pull request by quaxsze**
fix https://github.com/etalab/data.gouv.fr/issues/522

### CODE REVIEW

**Comment by geoffreyaldebert** [comment]
can we add also `resources` properties from `dataset` ? It would be very useful 😇

File: udata/core/dataset/search.py [hunk 1]
```diff
@@ -86,6 +86,18 @@ def serialize(cls, dataset):
...
 
+        serialized_resources = []
+        for resource in dataset.resources:
+            serialized_resources.append({
+                'id': resource.id,
+                'url': resource.url,
+                'format': resource.format,
+                'title': resource.title,
+                'schema': resource.schema,
+                'latest': resource.latest
```
[the remark this answers is not in the document]
Comment (geoffreyaldebert) [reply udata/core/dataset/search.py:97]: I think you cover the essentials properties, thx.

[the remark this answers is not in the document]
Comment (geoffreyaldebert) [reply udata/core/dataset/search.py:97]: filesize could be useful also if trustable. And beware, it's `filetype` not `file_type`

File: udata/core/dataset/search.py [hunk 2]
```diff
@@ -82,9 +82,29 @@ def serialize(cls, dataset):
...
             'owner': str(owner.id) if owner else None,
-            'format': [r.format.lower() for r in dataset.resources if r.format]
+            'format': [r.format.lower() for r in dataset.resources if r.format],
+            'extras': dataset.extras
         }
 
+        serialized_resources = []
+        for resource in dataset.resources:
+            serialized_resources.append({
+                'id': resource.id,
```
Comment (maudetes) [inline udata/core/dataset/search.py:92]:
```suggestion
                'id': str(resource.id),
```

File: udata/core/dataset/search.py [hunk 3]
```diff
@@ -82,9 +82,29 @@ def serialize(cls, dataset):
...
+                'url': resource.url,
+                'format': resource.format,
+                'title': resource.title,
+                'schema': resource.schema,
```
Comment (maudetes) [inline udata/core/dataset/search.py:96]: We'll need to read this information in the consumer. We could also put it at the dataset level, like we deed for format. What do you think?

**Review by maudetes — COMMENTED** [review]
Thank you for this PR!

Some extras entries could be `datetime`, which can't be serialized.

File: CHANGELOG.md [hunk 4]
```diff
@@ -18,6 +18,7 @@
   - The ModelAdapter, SearchQuery and SearchResult patterns were kept but heavily refactored.
   - Udata uses a Kafka producer to send documents to index to the search service.
   - Udata uses HTTP request to query the search service.
+- Add the `extras` field of Dataset, Reuse and Organization into their corresponding `ModelSearchAdapter` serialize method. [#2703](https://github.com/opendatateam/udata/pull/2703)
```
Comment (maudetes) [inline CHANGELOG.md:21]: I think we'll need to refactor this Changelog after when merging SearchDev.

File: udata/core/dataset/search.py [hunk 5]
```diff
@@ -82,9 +82,30 @@ def serialize(cls, dataset):
             'resources_count': len(dataset.resources),
             'organization': organization,
             'owner': str(owner.id) if owner else None,
-            'format': [r.format.lower() for r in dataset.resources if r.format]
+            'format': [r.format.lower() for r in dataset.resources if r.format],
+            'schema': [r.schema.get('name') for r in dataset.resources if r.schema],
+            'extras': dataset.extras
```
Comment (maudetes) [inline udata/core/dataset/search.py:87]: Extras could be of datetime type, which can't be serialized.

**Review by maudetes — APPROVED** [review]
I think we're good :)

**Pull request merged**